lang/mrustc/Portfile

```diff
@@ -38,67 +40,325 @@ use_configure   no
 #   depends_run-append port:[string map {"macports-gcc-" "gcc"} ${configure.compiler}]
 #}
 
-# https://github.com/thepowersgang/mrustc/pull/192
-patchfiles-append patch-mrustc-macos-bootstrap.diff
-
-# Set the default MRUSTC_LIBDIR
-patchfiles-append patch-mrustc-macports-libdir.diff
-post-patch {
-    reinplace "s|@MACPORTS_MRUSTC_LIBDIR@|${prefix}/lib/mrustc-${rust_version_major}|" ${worksrcpath}/src/main.cpp
-}
+# Patches available as git branch: https://github.com/catap/mrustc/tree/macports-1.39.0
+patch.pre_args  -p1
+patchfiles      0001-Fixes-for-bootstrapping-1.39-on-macOS.patch \
+                0002-Added-LINKFLAGS_EXTRA.patch \
+                0003-run_rustc-respect-j-option.patch \
+                0004-Added-LLVM_CMAKE_OPTS_EXTRA.patch \
+                0005-Added-wasm32-target.patch \
+                0006-Let-MacPorts-set-a-default-value-for-MRUSTC_LIBDIR.patch \
+                0007-Let-MacPorts-define-the-default-compiler.patch \
+                0008-Let-MacPorts-customize-the-default-Rust-target-versi.patch
 
-# Set the default CC
-patchfiles-append patch-mrustc-macports-compiler.diff
 post-patch {
+    # Set the default MRUSTC_LIBDIR
+    reinplace "s|@MACPORTS_MRUSTC_LIBDIR@|${prefix}/lib/mrustc|" ${worksrcpath}/src/main.cpp
+    # Set the default CC
     reinplace "s|@MACPORTS_MRUSTC_CC@|${configure.cc}|" ${worksrcpath}/src/trans/codegen_c.cpp
-}
-
-# Set the default gTargetVersion
-patchfiles-append patch-mrustc-macports-version.diff
-post-patch {
+    # Set the default gTargetVersion
     reinplace "s|@MACPORTS_MRUSTC_TARGET_VER@|${rust_version_major_underscore}|" ${worksrcpath}/src/main.cpp
-}
 
-# Patch the Rust source code with the patch included with mrustc
-post-patch {
+    # Patch the Rust source code with the patch included with mrustc
     system -W ${workpath}/rustc-${rust_version}-src "patch -p0 < ${worksrcpath}/rustc-${rust_version}-src.patch"
     file rename ${workpath}/rustc-${rust_version}-src ${worksrcpath}
+    # prevent it from re-download src
+    touch ${worksrcpath}/rustc-${rust_version}-src/dl-version
 }
 
-build {
-    system -W ${worksrcpath} "/usr/bin/env CXX=${configure.cxx} CC=${configure.cc} CXXFLAGS_EXTRA='${configure.cxxflags}' ${build.cmd} all"
-    system -W ${worksrcpath}/tools/minicargo "/usr/bin/env CXX=${configure.cxx} CC=${configure.cc} ${build.cmd} all"
-    system -W ${worksrcpath} "/usr/bin/env CXX=${configure.cxx} CC=${configure.cc} RUSTC_VERSION=${rust_version} RUSTC_TARGET=${build_arch}-apple-darwin MRUSTC_LIBDIR=output-${rust_version} MRUSTC_TARGET_VER=${rust_version_major} ${build.cmd} -f minicargo.mk LIBS"
+depends_build-append \
+                port:cmake \
+                port:git \
+                port:pkgconfig
+
+depends_lib-append \
+                port:curl \
+                port:libgit2 \
+                port:nghttp2 \
+                port:zlib
+
+# use the system python27 if present
+if {${os.platform} eq "darwin" && ${os.major} >= 11} {
```

Comment:
The latest version of macOS Monterey removed python, not only python3 gets made available when command line tools are installed.

Reply:
@Gcenx I'm using macOS 12.1 on my laptop and it has `python2.7`:
```
√ macports-ports % ls -l /usr/bin/python2.7
lrwxr-xr-x  1 root  wheel  75 Dec  8 00:39 /usr/bin/python2.7 -> ../../System/Library/Frameworks/Python.framework/Versions/2.7/bin/python2.7
√ macports-ports %
```

Reply:
@catap once you update your system that will be gone, it’s gotten a lot of people complaining online.